examples/plot_callosal_tract_profile.py

```diff
@@ -0,0 +1,558 @@
+"""
+==============================
+Plotting Novel Tract Profiles:
+==============================
+
+The following is an example of tractometry for a novel bundle and plotting the
+resulting FA tract profile. We will run tractometry for the *anterior forceps*
+using waypoint ROIs.
+
+**AFQ Waypoint ROI Tractometry:**
+
+.. note::
+
+  This example uses the Yeatman et al. waypoint ROI approach, first
+  described in [Yeatman2012]_ and further elaborated in [Yeatman2014]_.
+
+  The waypoint ROIs used in this example are from the anterior frontal lobe of
+  the corpus callosum (AntFrontal). The waypoint ROIs are from the human corpus
+  callosum templates:
+
+    https://figshare.com/articles/Templates_for_Automated_Fiber_Quantification_of_corpus_callosum_from_Diffusion_MRI_data/3381523
+
+"""
+import os.path as op
+import matplotlib.pyplot as plt
+import numpy as np
+import nibabel as nib
+import dipy.data as dpd
+from dipy.data import fetcher
+from dipy.io.streamline import save_tractogram, load_tractogram
+from dipy.stats.analysis import afq_profile, gaussian_weights
+from dipy.io.stateful_tractogram import StatefulTractogram
+from dipy.io.stateful_tractogram import Space
+
+import AFQ.data as afd
+import AFQ.tractography as aft
+import AFQ.registration as reg
+import AFQ.models.dti as dti
+import AFQ.segmentation as seg
+from AFQ.utils.volume import patch_up_roi
+
+import logging
+import sys
+
+# Ensure segmentation logging information is included in this example's output
+root = logging.getLogger()
+root.setLevel(logging.ERROR)
+logging.getLogger('AFQ.Segmentation').setLevel(logging.INFO)
+logging.getLogger('AFQ.tractography').setLevel(logging.INFO)
+handler = logging.StreamHandler(sys.stdout)
+handler.setLevel(logging.INFO)
+root.addHandler(handler)
+
+# Monkey patch the progress bars so don't pollute this example's output
+import AFQ._fixes as fixes
+
+def notqdm(iterable, *args, **kwargs):
+    """
+    replacement for tqdm that just passes back the iterable to silence `tqdm`
+    """
+    return iterable
+
+
+afd.tqdm = notqdm
+fixes.tqdm = notqdm
+seg.tqdm = notqdm
+
+
+def update_progressbar(progress, total_length):
+    return
+
+
+fetcher.update_progressbar = update_progressbar
+
+
+# Target directory for this example's output files
+working_dir = "./callosal_tract_profile"
+
+##########################################################################
+# Helper functions:
```

Comment:
Quality Control Report
- Consider incorporating anatomical slices and density map functionality as part of quality control report (from @36000)
- Consider adding a visualization that builds on the streamline density maps, which plots contours for each individual in MNI space (from @jyeatman)
- Consider placing ROIs and streamline density maps on anatomical image (from @jyeatman and @arokem)
- Consider plotting ODF (orbital distribution function) as part of QC report (from @richford)

NOTE as far as I know the quality control report functionality does not exist yet, but perhaps there is an existing enhancement request which can attach these additional enhancements.

Reply:
There is also some overlap with whether these graphics should be integrated with the pyAFQ pipeline.

NOTE In my mind this might be something to consider in tandem with the work on nipype integration

Reply:
For integration with the AFQ object, the general layout would look like this, with create_density_map as an example:

// creates a density map for this row if it does not already exist
// remember a row is a specific session for a specific subject
// should be a wrapper around the create_density_map function that handles filenames / meta json file / checking if the file exists
def _create_density_map(self, row): 

def get_create_density_map(): // calls _create_density_map  for every row and returns the resulting files as a list

def set_create_density_map(): // calls _create_density_map  for every row

Most AFQ object functions follow this format. Some have some functionality that should probably be extracted out into their own functions. Also I bet we could reduce a lot of code by making decorators for these types of functions. And, if the get / set functions are not documented somewhere, they should be.

Reply:
Addressed in #502